### PR TITLE
cargo fmt

### PR DESCRIPTION
**Pull request by sornas**
```
 benches/sylt_benchmark.rs |   6 +-
 build.rs                  |   1 -
 src/compiler.rs           | 402 ++++++++++++++++++++++++++++------------------
 src/error.rs              |  70 +++++---
 src/lib.rs                | 231 +++++++++++++-------------
 src/main.rs               |  11 +-
 src/sectionizer.rs        |  95 ++++++-----
 src/tokenizer.rs          |   2 +-
 src/vm.rs                 | 298 ++++++++++++++++++++++------------
 9 files changed, 650 insertions(+), 466 deletions(-)

```

### CODE REVIEW

File: benches/sylt_benchmark.rs [hunk 1]
```diff
@@ -5,8 +5,9 @@ use std::path::Path;
 macro_rules! bench_file {
     ( $name:ident ) => {
         pub fn $name(c: &mut Criterion) {
-            let prog = fs::read_to_string(Path::new(&format!("progs/bench/{}.sy", stringify!($name))))
-                .unwrap();
+            let prog =
```
Comment (FredTheDino) [inline benches/sylt_benchmark.rs:8]: :-1:

File: src/lib.rs [hunk 2]
```diff
@@ -158,15 +159,12 @@ impl PartialEq for Type {
             (Type::Float, Type::Float) => true,
             (Type::Bool, Type::Bool) => true,
             (Type::String, Type::String) => true,
-            (Type::Tuple(a), Type::Tuple(b)) => {
-                a.iter().zip(b.iter()).all(|(a, b)| a == b)
-            }
-            (Type::Union(a), b) | (b, Type::Union(a)) => {
-                a.iter().any(|x| x == b)
-            }
+            (Type::Tuple(a), Type::Tuple(b)) => a.iter().zip(b.iter()).all(|(a, b)| a == b),
```
Comment (FredTheDino) [inline src/lib.rs:162]: :question:

File: src/lib.rs [hunk 3]
```diff
@@ -158,15 +159,12 @@ impl PartialEq for Type {
...
+            (Type::Union(a), b) | (b, Type::Union(a)) => a.iter().any(|x| x == b),
```
Comment (FredTheDino) [inline src/lib.rs:163]: :question:

**Comment by sornas** [comment]
Uhhhhhhhhhhhh

**Comment by sornas** [comment]
Above comments look good

**Pull request merged**